packages/yoroi-extension/app/api/ergo/lib/state-fetch/remoteFetcher.js

```diff
@@ -1,45 +1,120 @@
-  HistoryFunc, HistoryRequest, HistoryResponse,
-  BestBlockRequest, BestBlockResponse,
-  SignedRequest, SignedResponse,
-  AssetInfoRequest, AssetInfoResponse,
+  AddressUtxoRequest,
+  AddressUtxoResponse,
+  AssetInfoRequest,
+  AssetInfoResponse,
+  BestBlockRequest,
+  BestBlockResponse,
+  HistoryFunc,
+  HistoryRequest,
+  HistoryResponse,
+  SignedRequest,
+  SignedResponse,
+  TxBodiesRequest,
+  TxBodiesResponse,
+  UtxoSumRequest,
+  UtxoSumResponse,
 } from './types';
 import type {
-  FilterFunc, FilterUsedRequest, FilterUsedResponse,
+  FilterFunc,
+  FilterUsedRequest,
+  FilterUsedResponse,
 } from '../../../common/lib/state-fetch/currencySpecificTypes';
 
 import type { IFetcher } from './IFetcher';
 
 import axios from 'axios';
+import { Logger, stringifyError } from '../../../../utils/logging';
 import {
-  Logger,
-  stringifyError
-} from '../../../../utils/logging';
-import {
+  CheckAddressesInUseApiError,
+  GetAssetInfoApiError,
+  GetBestBlockError,
+  GetTxHistoryForAddressesApiError,
   GetTxsBodiesForUTXOsApiError,
   GetUtxosForAddressesApiError,
   GetUtxosSumsForAddressesApiError,
-  GetTxHistoryForAddressesApiError,
-  GetBestBlockError,
-  SendTransactionApiError,
-  GetAssetInfoApiError,
-  CheckAddressesInUseApiError,
   InvalidWitnessError,
   RollbackApiError,
+  SendTransactionApiError,
 } from '../../../common/errors';
 
 import type { ConfigType } from '../../../../../config/config-types';
+import { fixUtxoToStringValues } from '../../index';
 
 import { decode, encode } from 'bs58';
+import JSONBigInt from 'json-bigint';
+import cloneDeep from 'lodash/cloneDeep'
+import BigNumber from 'bignumber.js';
 
 // populated by ConfigWebpackPlugin
 declare var CONFIG: ConfigType;
 
+function errClass(Err: Function): (* => Function) {
+  return _ => new Err();
+}
+
+type AxiosGet<T> = {|
+  responseMapper?: Object => T,
+  callerName: string,
+  errorFactory: (error: *) => Function,
+|};
+type AxiosPost<T> = {|
+  data?: *,
+  ...AxiosGet<T>,
+|};
+type AxiosParams<T> = AxiosGet<T> | AxiosPost<T>;
+type AxiosMethod = 'get' | 'post';
+type AxiosFunc<T> = (string, AxiosParams<T>) => Promise<T>
+
+function axiosRequest<T>(fetcher: RemoteFetcher, method: AxiosMethod): AxiosFunc<T> {
+  return (url, params) => {
+    const debug = (s: string, p: *) => {
+      // eslint-disable-next-line no-console
+      console.debug(`AXIOS[${method}][${url}] ${s} > `, cloneDeep(p));
+    };
+    debug('CALLING', params);
+    return axios(url, {
+      method,
+      timeout: 2 * CONFIG.app.walletRefreshInterval,
+      headers: {
+        'yoroi-version': fetcher.getLastLaunchVersion(),
+        'yoroi-locale': fetcher.getCurrentLocale()
+      },
+      transformResponse: resp => {
+        debug('RSPRAW', resp);
+        return JSONBigInt.parse(resp);
+      },
+      transformRequest: data => {
+        if (!data) {
+          return data;
+        }
+        const res = JSONBigInt.stringify(data);
```

Comment:
Since you pass the request as a string to axios, axios defaults the "content-type" header to "application/x-www-form-urlencoded", instead of "application/json" as desired. I find that adding this line at the global scope fixes it:
```
axios.defaults.headers.post['Content-Type'] = 'application/json';
```

Reply:
That's a good point! But I am kinda anxious changing the axios defaults as I am not sure if it will not accidentally affect some other parts of the system that might incidentally or intentionally rely on the axios type detecting, or something like that.

I have found that we can set the content type for that specific request every time in the same request-mapper tho, so fixed in https://github.com/Emurgo/yoroi-frontend/pull/2246/commits/64b72daf5aec865594343584cd9c1b2093c4ec07

Reply:
make sense!